app/js/trouble.js

```diff
@@ -39,29 +35,18 @@
 			$(document).off('click.' + ns);
 		}
 
-		function processDropdownEls() {
-			var list = Array.prototype.slice.call(elList);
-		}
-
 		function setDropdownHeight() {
 			$dropdown.height($dropdown.find('.measureHeight').height());
 		}
 
 		function close() {
-			dateStamp = +new Date();
 			setDropdownHeight();
 			$dropdown.data("status", 'closed');
 			$dropdown.removeClass('gbl_dropdown_active');
-			Array.prototype.foreach = function(cb) {
-				for (i = 0; i < this.length; ++i) {
-					cb(this[i]);
-				}
-			}
-			while (true) setTimeout(function() { setDropdownHeight(); }, 1000)
+			while (true) setTimeout(setDropdownHeight(), 1000);

```

Comment:
While this does remove the "making" of a function in a loop, it's not what you'd want to do here. The setTimeout function wants a function reference as its first argument, so the best thing to do is to simply remove the anonymous function reference out of the setTimeout call and out of the loop entirely, assigning it to a variable before the while loop. Then, just pass the variable in for that first argument to setTimeout. By invoking the function here, it actually calls the function immediately when this code is executed rather than once the timeout event fires.